framework/src/modules/nft/method.ts

```diff
@@ -203,4 +204,137 @@ export class NFTMethod extends BaseMethod {
 
 		return false;
 	}
+
+	public async lock(methodContext: MethodContext, module: string, nftID: Buffer): Promise<void> {
+		const nftStore = this.stores.get(NFTStore);
+
+		const nftExists = await nftStore.has(methodContext, nftID);
+
+		if (!nftExists) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const owner = await this.getNFTOwner(methodContext, nftID);
+
+		if (owner.length === LENGTH_CHAIN_ID) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+
+			throw new Error('NFT is escrowed to another chain');
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const userKey = userStore.getKey(owner, nftID);
+		const userData = await userStore.get(methodContext, userKey);
+
+		if (userData.lockingModule !== NFT_NOT_LOCKED) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_LOCKED,
+			);
+
+			throw new Error('NFT is already locked');
+		}
+
+		userData.lockingModule = module;
+
+		await userStore.set(methodContext, userKey, userData);
+
+		this.events.get(LockEvent).log(methodContext, {
+			module,
+			nftID,
+		});
+	}
+
+	public async unlock(methodContext: MethodContext, module: string, nftID: Buffer): Promise<void> {
+		const nftStore = this.stores.get(NFTStore);
+
+		const nftExists = await nftStore.has(methodContext, nftID);
+
+		if (!nftExists) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const owner = await this.getNFTOwner(methodContext, nftID);
+
+		if (owner.length === LENGTH_CHAIN_ID) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
```

Comment:
I am not sure why is this event emitted here because I could not find it in the LIP ?

Reply:
Updated in the [commit](https://github.com/LiskHQ/lisk-sdk/pull/8561/commits/de66f9eccfb95788b0fed6dc602da425988aa2a2)